plans/main.fmf

```diff
@@ -0,0 +1,59 @@
+summary: Run Python Podman Tests
+
+discover:
+    how: fmf
+execute:
+    how: tmt
+prepare:
+  - name: pkg dependencies
+    how: install
+    package:
+      - make
+      - podman
+      - python3-pip
+
+  - name: pip dependencies
+    how: shell
+    script:
+      - pip3 install -r test-requirements.txt
+
+  - name: ssh configuration
+    how: shell
+    script:
+        - ssh-keygen -t ecdsa -b 521 -f /root/.ssh/id_ecdsa -P ""
+        - cp /root/.ssh/authorized_keys /root/.ssh/authorized_keys%
+        - cat /root/.ssh/id_ecdsa.pub >>/root/.ssh/authorized_keys
+
+/upstream:
```

Comment:
unless you need to run all the below test plans in parallel, you can just use `filter: tag:upstream` and have this plan specification simplified by a lot I guess.

Reply:
yup.

now I am trying to simplify the matrix a little bit, but I still didn't figured how.

so we have 4 runs to do.

- `make lint`: runs the linting. we want to run this on fedora stable only
- `make integration`: coverage for integration tests, upstream only and ideally fedora stable only
- `make unittest`: coverage for unit tests, upstream only and fedora stable only
- `make tests`: will run all integration and unit tests. this requires python3.9->3.13 to be installed and should run everywhere

Reply:
and I am so confused if I should tag the tests or rather divide the tests virtually.

Reply:
I'd say let's go with what you find to be simplest. Also, feel free to enable just 1 environment `fedora-latest-stable` for now. I'll be happy to merge with just that 1 env :) .

You can add other distros and versions in followups.